rust/otap-dataflow/crates/engine/src/error.rs

```diff
@@ -6,30 +6,26 @@
 //! Important note: It is important not to use `!Send` data types in errors (e.g. avoid using Rc) to
 //! ensure these errors can be emitted in both `Send` and `!Send` contexts.
 
-use crate::control::{NodeControlMsg, PipelineControlMsg};
 use crate::node::{NodeId, NodeName};
 use otap_df_channel::error::SendError;
 use otap_df_config::{PortName, Urn};
 use std::borrow::Cow;
 
-/// All errors that can occur in the pipeline engine infrastructure.
+/// All errors that can occur in the pipeline engine infrastructure
+/// that contain a variant type <T>. Generally these errors are
+/// returned in situations where you may want to take ownership of the
+/// request after it has failed to send. Most callers that encounter
+/// ErrorT<T> and wish to drop the <T> can use .map_err(|e| e.into())
+/// to return an Error.
 #[derive(thiserror::Error, Debug)]
-pub enum Error<T> {
-    /// A wrapper for the config errors.
-    #[error("A config error occurred: {0}")]
-    ConfigError(#[from] Box<otap_df_config::error::Error>),
-
-    /// A wrapper for the channel errors.
-    #[error("A channel error occurred: {0}")]
-    ChannelRecvError(#[from] otap_df_channel::error::RecvError),
-
+pub enum ErrorT<T> {
```

Comment:
I don't have a better name to suggest for now, but if we find a better one we shouldn't hesitate :-)

Reply:
Renamed `TypedError`.